sdk/src/keystore/mod.rs

```diff
@@ -210,43 +329,69 @@ pub trait Backend {
     fn iter_bls_bn254(&self) -> impl Iterator<Item = bn254::Public>;
 }
 
+/// A convenience trait to extend the [`Backend`] trait with additional methods
+/// that provide convenient access to keys
 pub trait BackendExt: Backend {
-    fn ecdsa_key(&self) -> Result<subxt_signer::ecdsa::Keypair, Error> {
+    #[cfg(any(feature = "std", feature = "wasm"))]
+    fn ecdsa_key(&self) -> Result<TanglePairSigner<sp_core_subxt::ecdsa::Pair>, Error> {
         let first_key = self
             .iter_ecdsa()
             .next()
-            .ok_or_else(|| Error::Ecdsa("No ECDSA keys found".into()))?;
+            .ok_or_else(|| str_to_std_error("No ECDSA keys found"))?;
```

Comment:
Is there a reason to convert these all to `std::io::Error`?

Reply:
We should consider specific errors here.